### PR TITLE
add DeploymentSettings CRD

### PR DESCRIPTION
**Pull request by zreigz**
Add DeploymentSettings CRD + operator

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

File: controller/api/v1alpha1/deploymentsettings_types.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	// Create bindings.
+	// +kubebuilder:validation:Optional
+	Create []Binding `json:"create,omitempty"`
+
+	// Git bindings.
+	// +kubebuilder:validation:Optional
+	Git []Binding `json:"git,omitempty"`
+}
+
+type StackSettings struct {
```
Comment (michaeljguarino) [inline controller/api/v1alpha1/deploymentsettings_types.go:80]: We should also add `ConnectionRef` here mapping to `connectionId` in the stack settings

File: controller/internal/controller/deploymentsettings_controller.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+func (r *DeploymentSettingsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	logger := log.FromContext(ctx)
+
+	settings := &v1alpha1.DeploymentSettings{}
+	if err := r.Get(ctx, req.NamespacedName, settings); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// make sure there is only one CRD object with the `global` name and the agent namespace
+	if settings.Name != deploymentSettingsName || settings.Namespace != deploymentSettingsNamespace {
```
Comment (zreigz) [inline controller/internal/controller/deploymentsettings_controller.go:67]: @michaeljguarino I've added such a check to make sure we have only one DeploumentSettings CRD object
with the `global` name and the agent namespace `plrl-deploy-operator`.

File: controller/api/v1alpha1/deploymentsettings_types.go (hunk 1 above)
Comment (zreigz) [reply controller/api/v1alpha1/deploymentsettings_types.go:81]: done

File: controller/internal/controller/deploymentsettings_controller.go (hunk 2 above)
Comment (michaeljguarino) [reply controller/internal/controller/deploymentsettings_controller.go:67]: Ok cool that's probably wise

**Pull request merged**